tools/autograph-client/client.go

```diff
@@ -282,16 +311,31 @@ func verifyContentSignature(input []byte, resp signatureresponse, endpoint strin
 	return ecdsa.Verify(pubKey, input, sig.R, sig.S)
 }
 
-func verifyXPI(input []byte, resp signatureresponse) bool {
-	sig, err := xpi.Unmarshal(resp.Signature, input)
-	if err != nil {
-		log.Fatal(err)
-	}
-	err = sig.VerifyWithChain(nil)
-	if err != nil {
-		log.Fatal(err)
+func verifyXPI(input []byte, req signaturerequest, resp signatureresponse, reqType requestType) bool {
+	switch reqType {
+	case requestTypeData:
+		sig, err := xpi.Unmarshal(resp.Signature, input)
+		if err != nil {
+			log.Fatal(err)
+		}
+		err = sig.VerifyWithChain(nil)
+		if err != nil {
+			log.Fatal(err)
+		}
+		return true
+	case requestTypeFile:
+		signedFile, err := base64.StdEncoding.DecodeString(resp.SignedFile)
```

Comment:
You already call `DecodeString` earlier

Reply:
? I don't see it. There's another call on resp.SignedFile afterwards.

Reply:
nevermind, misread your code the first time